R/tax_unique.R

```diff
@@ -262,7 +267,24 @@ tax_unique <- function(occdf = NULL, binomial = NULL, species = NULL,
   }
 
   #Remove absolute repeats
-  occurrences <- unique(occurrences)
+  if (orig) {
+    occurrences_with_dupes <- occurrences
+    # List the rows of the original dataframe that correspond to each unique row
+    # Hack to include NAs as groups
+    occurrences_with_dupes[is.na(occurrences_with_dupes)] <- "thisisanNAvalue"
+    occurrences_with_dupes$rows <- seq_len(nrow(occurrences_with_dupes))
+    form <- as.formula(
+      paste(". ~",
+            paste(c(rev(higher_names), "genus", "genus_species"),
```

Comment:
I think this is the line causing issues with resolution... "genus_species" will not always be present in the dataset?